perfkitbenchmarker/providers/rackspace/rackspace_network.py

```diff
@@ -13,140 +13,295 @@
+  """An object representing a Rackspace Network."""
+
+  CLOUD = providers.RACKSPACE
+
+  def __init__(self, network_spec):
+    super(RackspaceNetwork, self).__init__(network_spec)
+    self.tenant_id = network_spec.tenant_id
+    name = FLAGS.rackspace_network_name or 'pkb-network-%s' % FLAGS.run_uri
+    self.network_resource = RackspaceNetworkResource(name, self.tenant_id)
+    self.subnet = RackspaceSubnet(self.network_resource.id, DEFAULT_SUBNET_CIDR,
+                                  ip_ver='4', name='subnet-%s' % name,
+                                  tenant_id=self.tenant_id)
+    self.security_group = RackspaceSecurityGroup('default-internal-%s' % name)
+    self.default_firewall_rules = []
+
+  @staticmethod
+  def _GetNetworkSpecFromVm(vm):
+    return RackspaceNetworkSpec(tenant_id=vm.tenant_id, region=vm.zone)
+
+  @classmethod
+  def _GetKeyFromNetworkSpec(cls, spec):
+    return (cls.CLOUD, spec.tenant_id, spec.region)
+
+  def Create(self):
+    if FLAGS.rackspace_network_name is None:
+      self.network_resource.Create()
+      self.subnet.Create()
+      self.security_group.Create()
+      self.default_firewall_rules = self._GenerateDefaultRules(
+          self.security_group.id, self.network_resource.name)
+      for rule in self.default_firewall_rules:
+        rule.Create()
+
+  def Delete(self):
+    if FLAGS.rackspace_network_name is None:
+      for rule in self.default_firewall_rules:
+        rule.Delete()
+      self.security_group.Delete()
+      self.subnet.Delete()
+      self.network_resource.Delete()
+
+  def _GenerateDefaultRules(self, sec_group_id, network_name):
+    firewall_rules = [
+        RackspaceSecurityGroupRule(
+            sec_group_rule_name='tcp-default-internal-%s' % network_name,
+            sec_group_id=sec_group_id,
+            direction=INGRESS,
+            ip_ver=IPV4,
+            protocol=TCP),
+        RackspaceSecurityGroupRule(
+            sec_group_rule_name='udp-default-internal-%s' % network_name,
+            sec_group_id=sec_group_id,
+            direction=INGRESS,
+            ip_ver=IPV4, protocol=UDP),
+        RackspaceSecurityGroupRule(
+            sec_group_rule_name='icmp-default-internal-%s' % network_name,
+            sec_group_id=sec_group_id,
+            direction=INGRESS,
+            ip_ver=IPV4, protocol=ICMP)]
+    return firewall_rules
+
+
+class RackspaceFirewall(network.BaseFirewall):
+  """An object representing a Rackspace Security Group applied to PublicNet and
+  ServiceNet."""
+
+  CLOUD = providers.RACKSPACE
+
+  def __init__(self):
+    # TODO(meteorfox) Support a Firewall per region
+    self._lock = threading.Lock()  # Guards security-group creation/deletion
+    self.firewall_rules = {}
+
+
+  def AllowPort(self, vm, port, to_port=None, source_range=None):
+    if vm.is_static or not FLAGS.use_security_group:
+      # At Rackspace all ports are open by default
+      return
+    # TODO(meteorfox) Implement security groups support
```

Comment:
Maybe raise `NotImplementedError` here (until you implement security groups support)?